.github/workflows/build-and-run.yml

```
@@ -0,0 +1,116 @@
+name: Build-And-Run Tests
+
+on: [pull_request, push]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
+
+jobs:
+  build_and_test:
+    env:
+      ALL_EXCLUDE: stage_dem|pybind\.unwrap\.phass
```

Last but not least, let's add a comment here to explain why we're excluding these particular tests, because otherwise I will 100% forget why we did things this way :grin:

```suggestion
      # stage_dem requires AWS permissions to access the S3 bucket containing the DEM data.
      # pybind.unwrap.phass occasionally segfaults on GHA runners.
      ALL_EXCLUDE: stage_dem|pybind\.unwrap\.phass
```